### ISSUE TITLE
Snapshots failing in version 2.14.2

### ISSUE THREAD

**Issue by cult-hero13** [github/backup-utils#433]
After updating backup-utils from version 2.11.0 to 2.14.2, my backups started failing with an error for ghe-ssh.  This was due to the change in the file from "sha256sum" to "git hash-object --stdin".  I simply installed git on the backup server and backups would run.

The backups still failed though, with the following message:

```
This port is reserved for git operations.
To administer the host over SSH, use port 122 with `ssh -p 122`.
rsync: connection unexpectedly closed (0 bytes received so far) [receiver]
rsync error: error in rsync protocol data stream (code 12) at io.c(600) [receiver=3.0.6]
```

The backup log started with:

```
Starting backup of <hostname> with backup-utils v2.14.2 in snapshot 20180830T063820
Connect <hostname>:22 OK (v2.14.3)
```

and ended with:

```
sent 85328 bytes  received 1229419074 bytes  10288739.77 bytes/sec
total size is 1917130257  speedup is 1.56
* Disabling ES index flushing ...
* Performing follow-up sync of ES indices ...
receiving incremental file list

sent 3947 bytes  received 89569 bytes  187032.00 bytes/sec
total size is 1917130257  speedup is 20500.56
* Enabling ES index flushing ...
Completed backup of <hostname>:22 in snapshot 20180830T070908 at 07:19:33
Error: Snapshot incomplete. Some steps failed: pages, storage.
```

My backup.config file had the following:

```
GHE_HOSTNAME="<hostname>"
GHE_DATA_DIR="<data directory>"
GHE_NUM_SNAPSHOTS=7
GHE_EXTRA_SSH_OPTS="-p 122 -i <path to github_id_rsa key file>"
```

I had been using these settings for many versions prior to this.  It turns out they are no longer valid, specifically the GHE_EXTRA_SSH_OPTS setting.  Once I removed "-p 122",  the logs started showing a proper connection to the github server over port 122, instead of 22, and the snapshots completed.

**Comment by lildude** [comment]
> After updating backup-utils from version 2.11.0 to 2.14.2, my backups started failing with an error for ghe-ssh. This was due to the change in the file from "sha256sum" to "git hash-object --stdin".

Yes, this is expected. We changed the requirements to include git in 2.13.0 as we chose to switch to it for checksum calculation used to generate unique multiplex filenames as sha256sum isn't available on all platforms and sometimes has a different name for the same functionality.  This change of requirements is highlighted right at the top of the README.

> I had been using these settings for many versions prior to this. It turns out they are no longer valid, specifically the GHE_EXTRA_SSH_OPTS setting. Once I removed "-p 122", the logs started showing a proper connection to the github server over port 122, instead of 22, and the snapshots completed.

🤔 I've just tested this myself and sure enough, I hit the same problem. I've dug into this and I see why this is happening and also why this is only affecting pages and storage.

PR on the way. Thanks for bringing this to our attention.
